src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/Completeness/SqlGetCompletenessFamilyMasks.php

```
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Enrichment\Bundle\Product\Query\Sql\Completeness;
+
+use Akeneo\Pim\Enrichment\Component\Product\Completeness\Model\CompletenessFamilyMask;
+use Akeneo\Pim\Enrichment\Component\Product\Completeness\Model\CompletenessFamilyMaskPerChannelAndLocale;
+use Akeneo\Pim\Enrichment\Component\Product\Completeness\Query\GetCompletenessFamilyMasks;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @author    Pierre Allard <pierre.allard@akeneo.com>
+ * @copyright 2019 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class SqlGetCompletenessFamilyMasks implements GetCompletenessFamilyMasks
+{
+    /** @var Connection */
+    private $connection;
+
+    /** @var array */
+    private $cache;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+        $this->cache = [];
+    }
+
+    /**
+     * @param string[] $familyCodes
+     *
+     * @return CompletenessFamilyMask[]
+     */
+    public function fromFamilyCodes(array $familyCodes): array
+    {
+        $result = [];
+        $nonFetchedFamilyCodes = [];
+        foreach ($familyCodes as $familyCode) {
+            if (isset($this->cache[$familyCode])) {
+                $result[$familyCode] = $this->cache[$familyCode];
+            } else {
+                $nonFetchedFamilyCodes[] = $familyCode;
+            }
+        }
+
+        $fetched = $this->fetch($nonFetchedFamilyCodes);
+        foreach ($fetched as $familyCode => $masks) {
+            $this->cache[$familyCode] = $masks;
+            $result[$familyCode] = $masks;
+        }
+
+        return $result;
+    }
+
+    private function fetch(array $familyCodes): array
```

what about familyCodes that does not exist? 
We could throw an exception.

---

Done, but I don't like that for queries, we already debate on that.

---

get vs find.

Should you find a family mask? I don't think so.

---

Because we name Get by default. But I do agree ;)